Same name imports with numbers in the directory structure generates invalid imports

Description
-------------

Same name imports with numbers in the directory structure generates invalid imports.

Mockery Version
--------
v2.44.1

Go Version
--------------
Go tip, 49adb8e46

Installation Method
-------------------

- [ ] Binary Distribution
- [ ] Docker
- [ ] brew
- [ ] go install
- [x] Other: manually built

Steps to Reproduce
------------------
With a file named `x.go`:
```
package x

import (
	a "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-02-01/containerservice"
	b "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-12-01/containerservice"
)

type Foo interface {
	A(a.Resource)
	B(b.Resource)
}
```
Call `mockery --all`.

### Expected Behavior
Valid Go code.

### Actual Behavior
The generated import alias begins with a digit, something that isn't allowed:
```
Between the lines is the file (mock.go) mockery generated in-memory but detected as invalid:
--------------------------------------------------------------------------------------------
// Code generated by mockery v2.44.1. DO NOT EDIT.

package mocks

import 2020_12_01containerservice "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-12-01/containerservice"
import containerservice "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-02-01/containerservice"
import mock "github.com/stretchr/testify/mock"
import x "x"

// Foo is an autogenerated mock type for the Foo type
type Foo struct {
	mock.Mock
}


// A provides a mock function with given fields: _a0
func (_m *Foo) A(_a0 containerservice.Resource)  {_m.Called(_a0)
}

// B provides a mock function with given fields: _a0
func (_m *Foo) B(_a0 2020_12_01containerservice.Resource)  {_m.Called(_a0)
}

// NewFoo creates a new instance of Foo. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewFoo(t interface {
	mock.TestingT
	Cleanup(func())
}) *Foo {
	mock := &Foo{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}

--------------------------------------------------------------------------------------------
Error walking Foo: mock.go:5:8: import path must be a string (and 1 more errors)
```

--- a/pkg/fixtures/12345678/http/http.go
+++ b/pkg/fixtures/12345678/http/http.go
@@ -1,0 +1,3 @@
+package http
+
+type MyStruct struct{}

--- a/pkg/fixtures/same_name_imports.go
+++ b/pkg/fixtures/same_name_imports.go
@@ -3,6 +3,7 @@ package test
 import (
 	"net/http"
 
+	number_dir_http "github.com/vektra/mockery/v2/pkg/fixtures/12345678/http"
 	my_http "github.com/vektra/mockery/v2/pkg/fixtures/http"
 )
 
@@ -10,4 +11,5 @@ import (
 type Example interface {
 	A() http.Flusher
 	B(fixtureshttp string) my_http.MyStruct
+	C(fixtureshttp string) number_dir_http.MyStruct
 }

--- a/pkg/generator.go
+++ b/pkg/generator.go
@@ -25,7 +25,7 @@ import (
 
 const mockConstructorParamTypeNamePrefix = "mockConstructorTestingT"
 
-var invalidIdentifierChar = regexp.MustCompile("[^[:digit:][:alpha:]_]")
+var invalidIdentifierChar = regexp.MustCompile("^[[:digit:]]|[^[:digit:][:alpha:]_]")
 
 func DetermineOutputPackageName(
 	interfaceFileName string,

--- a/pkg/generator_test.go
+++ b/pkg/generator_test.go
@@ -260,6 +260,7 @@ func (s *GeneratorSuite) TestGeneratorPrologueWithMultipleImportsSameName() {
 
 	expected := `package mocks
 
+import _2345678http "github.com/vektra/mockery/v2/pkg/fixtures/12345678/http"
 import fixtureshttp "github.com/vektra/mockery/v2/pkg/fixtures/http"
 import http "net/http"
 import mock "github.com/stretchr/testify/mock"

